pkg/cache/cache_test.go

```diff
@@ -69,185 +123,157 @@ var _ = Describe("Informer Cache", func() {
-			reader, err := cache.New(cfg, cache.Options{})
+			By("getting the Kubernetes service")
+			svc := &kcorev1.Service{}
+			svcKey := client.ObjectKey{Namespace: "default", Name: "kubernetes"}
+			Expect(informerCache.Get(context.Background(), svcKey, svc)).NotTo(HaveOccurred())
+
+			By("verifying that the returned service looks reasonable")
+			Expect(svc.Name).To(Equal("kubernetes"))
+			Expect(svc.Namespace).To(Equal("default"))
+		})
+
+		It("should support filtering by labels", func() {
+			By("listing pods with a particular label")
+			// NB: each pod has a "test-label" equal to the pod name
+			out := kcorev1.PodList{}
+			Expect(informerCache.List(context.TODO(), client.InNamespace(testNamespaceTwo).
+				MatchingLabels(map[string]string{"test-label": "test-pod-2"}), &out)).NotTo(HaveOccurred())
+
+			By("verifying the returned pods have the correct label")
+			Expect(out.Items).NotTo(BeEmpty())
+			Expect(len(out.Items)).To(Equal(1))
+			actual := out.Items[0]
+			Expect(actual.Labels["test-label"]).To(Equal("test-pod-2"))
+		})
+
+		It("should be able to list objects by namespace", func() {
+			By("listing pods in test-namespace-1")
+			listObj := &kcorev1.PodList{}
+			Expect(informerCache.List(context.Background(),
+				client.InNamespace(testNamespaceOne),
+				listObj)).NotTo(HaveOccurred())
+
+			By("verifying that the returned pods are in test-namespace-1")
+			Expect(listObj.Items).NotTo(BeEmpty())
+			Expect(len(listObj.Items)).To(Equal(1))
+			actual := listObj.Items[0]
+			Expect(actual.Namespace).To(Equal(testNamespaceOne))
+		})
+
+		It("should deep copy the object unless told otherwise", func() {
+			By("retrieving a specific pod from the cache")
+			out := &kcorev1.Pod{}
+			podKey := client.ObjectKey{Name: "test-pod-2", Namespace: testNamespaceTwo}
+			Expect(informerCache.Get(context.Background(), podKey, out)).NotTo(HaveOccurred())
+
+			By("verifying the retrieved pod is equal to a known pod")
+			Expect(out).To(Equal(knownPod2))
+
+			By("altering a field in the retrieved pod")
+			*out.Spec.ActiveDeadlineSeconds = 4
+
+			By("verifying the pods are no longer equal")
+			Expect(out).NotTo(Equal(knownPod2))
+		})
+
+		It("should error out for missing objects", func() {
+			By("getting a service that does not exists")
+			svc := &kcorev1.Service{}
+			svcKey := client.ObjectKey{Namespace: "unknown", Name: "unknown"}
+
+			By("verifying that an error is returned")
+			err := informerCache.Get(context.Background(), svcKey, svc)
+			Expect(err).To(HaveOccurred())
+			Expect(errors.IsNotFound(err)).To(Equal(true))
+		})
+	})
+
+	Describe("as an Informer", func() {
+		It("should be able to get informer for the object", func(done Done) {
+			By("getting an shared index informer for a pod")
+			pod := &kcorev1.Pod{
+				ObjectMeta: kmetav1.ObjectMeta{Name: "informer-obj", Namespace: "default"},
+				Spec:       kcorev1.PodSpec{Containers: []kcorev1.Container{{Name: "nginx", Image: "nginx"}}},
+			}
```

Comment:
IMO, squeezing above two values in single line impact the readability. I think making them multi-line will appear more verbose, but improves the readability.